[Workers] clarify bindings + envvars + secrets (#13205)
cloudflare/cloudflare-docs: content/workers/configuration/secrets.md

## Background

> Secrets are a type of binding that allow you to attach encrypted text values to your Worker. Secrets are not visible after they are set. Secrets are used for storing sensitive information like API keys and auth tokens. Secrets are available on the [`env` parameter](/workers/runtime-apis/handlers/fetch/#parameters) passed to your Worker's [`fetch` event handler](/workers/runtime-apis/handlers/fetch/).

Reviewer: Isn't saying that secrets "are not visible" a bit generic/ambiguous?

```suggestion
Secrets are a type of binding that allow you to attach encrypted text values to your Worker. Secrets are displayed in any manner after they are set and can only be accessed programmatically. Secrets are used for storing sensitive information like API keys and auth tokens. Secrets are available on the [`env` parameter](/workers/runtime-apis/handlers/fetch/#parameters) passed to your Worker's [`fetch` event handler](/workers/runtime-apis/handlers/fetch/).
```

Reviewer 2: Maybe "not visible in the dashboard"?

Reviewer: I was being generic because this applies to wrangler too:
![Screenshot 2024-02-28 at 21 50 48](https://github.com/cloudflare/cloudflare-docs/assets/61631103/cee0b3f2-10cb-419e-9515-63e47b898dc3)

Author: TBH, I don't see an issue with saying "secrets are not visible", the screenshot also says "(hidden)" which means it's not visible. "Not visible" or "hidden" seems technically accurate to me.

Author: How about "Secrets cannot be seen after they are set and can only be [accessed programmatically](/workers/wrangler/commands/#secret)."

Reviewer: > TBH, I don't see an issue with saying "secrets are not visible", the screenshot also says "(hidden)" which means it's not visible. "Not visible" or "hidden" seems technically accurate to me.

yeah I don't think it's super wrong, my only concern is that people could think that secrets are not visibile even programmatically... not that that would make too much sense... but better to avoid potential confusions I guess? 😅

> Secrets are a type of bindings that allow you to attach encrypted text values to your Worker. Secrets are not visible after they are set. Secrets are used for storing sensitive information like API keys and auth tokens. Secrets are available on the [`env` parameter](/workers/runtime-apis/handlers/fetch/#parameters) passed to your Worker's [`fetch` event handler](/workers/runtime-apis/handlers/fetch/).

Author: ```suggestion
Secrets are a type of binding that allow you to attach encrypted text values to your Worker. Secrets are not visible after they are set. Secrets are used for storing sensitive information like API keys and auth tokens. Secrets are available on the [`env` parameter](/workers/runtime-apis/handlers/fetch/#parameters) passed to your Worker's [`fetch` event handler](/workers/runtime-apis/handlers/fetch/).
```

> Secrets are a type of binding that allow you to attach encrypted text values to your Worker. You cannot see secrets after you set them and can only [access secrets programmatically](/workers/wrangler/commands/#secret). Secrets are used for storing sensitive information like API keys and auth tokens. Secrets are available on the [`env` parameter](/workers/runtime-apis/handlers/fetch/#parameters) passed to your Worker's [`fetch` event handler](/workers/runtime-apis/handlers/fetch/).

Author: ```suggestion
Secrets are a type of binding that allow you to attach encrypted text values to your Worker. You cannot see secrets after you set them and can only access secrets via [Wrangler](/workers/wrangler/commands/#secret) or programmatically via the [`env` parameter](/workers/runtime-apis/handlers/fetch/#parameters). Secrets are used for storing sensitive information like API keys and auth tokens. Secrets are available on the [`env` parameter](/workers/runtime-apis/handlers/fetch/#parameters) passed to your Worker's [`fetch` event handler](/workers/runtime-apis/handlers/fetch/).
```
